### ISSUE TITLE
Documentation examples to also include serializing to Rust examples

### ISSUE THREAD

**Issue by joshuataylor** [jorgecarleitao/arrow2#1039]
Hi!

For the arrays there is a great line about conceptualising how an Arrow concept fits into Rust, for example this [handy snippet](https://github.com/jorgecarleitao/arrow2/blob/main/src/array/utf8/mod.rs):

```
/// A [`Utf8Array`] is arrow's equivalent of an immutable `Vec<Option<String>>`.
/// Cloning and slicing this struct is `O(1)`.
```

Are there thoughts around also providing examples to convert this back to Rust `Vec<Option<String>>` (or whatever the context is) in an efficient/recommended way?

**Comment by jorgecarleitao** [comment]
Good point. I have been trying improve the docs to achieve that. In main we have something like that already:

https://github.com/jorgecarleitao/arrow2/blob/main/src/array/utf8/mod.rs#L37

but improvements are always welcome ^^

**Comment by joshuataylor** [comment]
Awesome!

I'll take notes about how I convert them and maybe write a guide, it would be handy for the getting started guide that exists.

**Comment by jorgecarleitao** [comment]
@jkcoxson, thanks for your interest!

Could you provide the function signature that you are looking for? (e.g. `fn f(chunk: Chunk, index: usize) -> Option<PrimitiveArray<i32>> {todo!()}`?) just so we get an idea of the documentation.

This because the arrow spec is quite extensive and it is a bit difficult to narrow this issue to a concrete usecase.

**Comment by joshuataylor** [comment]
Might be worth converting this into a discussion @jorgecarleitao :)
